include/libcyphal/types.hpp

```diff
@@ -149,6 +150,31 @@ CETL_NODISCARD UpVariant upcastVariant(Variant&& variant)
                        std::forward<Variant>(variant));
 }
 
+/// @brief Wraps the given action into a lambda and performs it without throwing exceptions.
+///
+/// In use f.e. for `cetl::visit` which might hypothetically throw an exceptions.
+///
+/// @return `true` if the action was performed successfully, `false` if an exception was thrown.
+///         Always `true` if exceptions are disabled.
+///
+template <typename Exception = std::exception, typename Action>
+CETL_NODISCARD bool performWithoutThrowing(Action&& action) noexcept
+{
+#if defined(__cpp_exceptions)
+    try
+    {
+#endif
+        std::forward<Action>(action)();
+        return true;
+#if defined(__cpp_exceptions)
+    } catch (const Exception& ex)
+    {
+        CETL_DEBUG_ASSERT(false, ex.what());
```

Comment:
Sorry but I was thinking this was distinct from the debug assert. Failures may be recoverable and this would lower the utility of debug assert which should be used for "this should never happen" cases only. Uh...just remove this line and ignore my previous request. Sorry.

Reply:
ok